bin/oneoff/backfill_data/backfill_section_course_id.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+# This script backfills the section table to add course_ids to old sections that have a script but no course.
+
+CDO.log = Logger.new($stdout)
+ActiveRecord::Base.record_timestamps = false
+
+options = {actually_update: false, start_with: 1}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-u', '--actually-update', 'Actually perform the update.') do
+    options[:actually_update] = true
+  end
+  opts.on('-sSTART', '--start-with=START', Integer, 'Start with section ID START.') do |s|
+    options[:start_with] = s
+  end
+  opts.on('-h', '--help', 'Add -u to perform the update.') do
+    puts opts
+    exit
+  end
+end.parse!
+CDO.log.info "Called with options: #{options}"
+
+BATCH_SIZE = 100
```

Comment:
for next time I'd suggest a batch size of around 10_000. main limits on the batch size are that the query doesn't timeout and your program doesn't run out of memory. many small reads with pagination / offsets can end up taking longer / putting more load on the DB then a few larger queries.